File: THPT/2024/DE KSHSG L3, KIM LIÊN- TIN/DE KSHSG L3/CODE/C++/ANHEM.cpp
#include <fstream>

int sum_of_divisors(int x) {
    int sum = 0;
    for (int i = 2; i <= x / 2; ++i) {
        if (x % i == 0) {
            sum += i;
        }
    }
    return sum;
}

bool kt_ae(int m, int n) {
    int sx = sum_of_divisors(m);
    int sy = sum_of_divisors(n);
    return (sx == sy && sx > 0);
}

int main() {
    std::ifstream f1("ANHEM.INP");
    std::ofstream f2("ANHEM.OUT");
    int m, n;
    f1 >> m >> n;
    if (kt_ae(m, n)) {
        f2 << "YES\n";
    } else {
        f2 << "NO\n";
    }
    return 0;
}

File: THPT/2024/DE KSHSG L3, KIM LIÊN- TIN/DE KSHSG L3/CODE/PYTHON/ANHEM.py
def sum_of_divisors(x):
    sx = 0
    for i in range(2, x // 2 + 1):
        if x % i == 0:
            sx += i
    return sx

def kt_ae(m, n):
    sx = sum_of_divisors(m)
    sy = sum_of_divisors(n)
    return sx == sy and sx > 0

def main():
    with open('ANHEM.INP', 'r') as f1, open('ANHEM.OUT', 'w') as f2:
        m, n = map(int, f1.readline().split())
        if kt_ae(m, n):
            f2.write('YES\n')
        else:
            f2.write('NO\n')

if __name__ == "__main__":
    main()
